MDANSE/Src/MDANSE/MolecularDynamics/Trajectory.py

```diff
@@ -54,11 +60,223 @@ def __init__(self, filename, trajectory_format=None):
         self._atom_cache = {}
+        self._selection = []
+        self.selection_getter = None
+        self._transmutation = {}
+        self._transmuted_types = []
+
+    @property
+    def atom_indices(self) -> Sequence[int]:
+        """Indices of the currently selected atoms."""
+        if self._selection:
+            return self._selection
+        return list(range(len(self.atom_types)))
+
+    @property
+    def atom_types(self) -> Sequence[str]:
+        """Chemical elements of ALL atoms, with transmutation applied."""
+        if self._transmuted_types:
+            return self._transmuted_types
+        if not self._transmutation:
+            return self._trajectory.chemical_system.atom_list
+        temp = copy.deepcopy(self._trajectory.chemical_system.atom_list)
+        for index, type in self._transmutation.items():
+            temp[index] = type
+        self._transmuted_types = temp
+        return self._transmuted_types
+
+    @property
+    def atom_names(self) -> Sequence[str]:
+        """Labels of ALL the atoms, after transmutation."""
+        if self._grouping_level in {"atom", "each atom"}:
+            if not self._element_from_label:
+                self._element_from_label = {element : element for element in self.unique_elements}
+            return self.atom_types
+        if not self._atom_names:
+            if self._grouping_level == "each molecule":
+                for mol_name, clusters in self.chemical_system._clusters.items():
+                    for mol_number, cluster in enumerate(clusters):
+                        if set(cluster).issubset(self.atom_indices):
+                            self._group_lookup[f"{mol_name}_mol{mol_number + 1}"] = cluster
+                self._atom_names = list(self._group_lookup.keys())
+            elif self._grouping_level == "molecule":
+                temp_names = {}
+                for mol_name in self.chemical_system._clusters:
+                    self._group_lookup.setdefault(mol_name, 0)
+                    for mol_number, cluster in enumerate(
+                        self.chemical_system._clusters[mol_name],
+                    ):
+                        if not set(cluster).issubset(self.atom_indices):
+                            continue
+                        self._group_lookup[mol_name] += len(cluster)
+                        for x in cluster:
+                            temp_names[x] = (f"<{mol_name}>/{self.atom_types[x]}")
+                self._element_from_label = {v: self.atom_types[k] for k, v in temp_names.items()}
+                self._atom_names = copy.deepcopy(self.atom_types)
+                for k, v in temp_names.items():
+                    self._atom_names[k] = v
+        return self._atom_names
+
+    @property
+    def element_from_label(self) -> dict[str, str]:
+        """Mapping from atom label to its chemical element name."""
+        if not self._element_from_label:
+            self.atom_names
```

Comment:
```suggestion
            return self.atom_names
```
Is this meant to have a `return`?